runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -516,7 +547,8 @@ static void *dereferenceStructPointerChain(void *baseStruct, TR::Node *baseNode,
 
                fieldAddress = curStruct + symRef->getOffset();
                }
-            else if (comp->getSymRefTab()->isImmutableArrayShadow(symRef))
+            else if (comp->getSymRefTab()->isImmutableArrayShadow(symRef) ||
+                     isBaseStableArray)
```

Comment:
It looks like this condition and the one in `verifyFieldAccess()` both assume that `isBaseStableArray` implies that `symRef` is an array shadow, but nothing has checked that it is. I think it would be good to check near the beginning of `transformIndirectLoadChainImpl()`, e.g.
```
if (isBaseStableArray && !symRef->getSymbol()->isArrayShadowSymbol())
   return false;
```

Reply:
Added the check.